In [1]:
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import pandas as pd

sns.set()
sns.set_context('talk')

In [2]:
import plotly.offline as py
import plotly.express as px

import plotly.graph_objs as go
import plotly.figure_factory as ff
import cufflinks as cf

cf.set_config_file(offline=True, world_readable=False)

In [2]:
import ipywidgets as widgets
from ipywidgets import interact, interactive, fixed, interact_manual
def df_interact(df, nrows=7, ncols=7):
    '''
    Outputs sliders that show rows and columns of df
    '''
    def peek(row=0, col=0):
        return df.iloc[row:row + nrows, col:col + ncols]

    row_arg = (0, len(df), nrows) if len(df) > nrows else fixed(0)
    col_arg = ((0, len(df.columns), ncols)
               if len(df.columns) > ncols else fixed(0))
    
    interact(peek, row=row_arg, col=col_arg)
    print('({} rows, {} columns) total'.format(df.shape[0], df.shape[1]))

def display_df(df, rows=pd.options.display.max_rows,
               cols=pd.options.display.max_columns):
    with pd.option_context('display.max_rows', rows,
                           'display.max_columns', cols):
        display(df)

## 책 데이터에 대한 분석

책 데이터는 22개 카테고리로 나뉘어져 저장되어 있다. 우선은 책 데이터가 잘 크롤되었는지 확인하려고 한다.

In [3]:
book1_1 = pd.read_csv("Data/book_1_1.csv")
book1_1

Unnamed: 0           ISBN  \
0   9791165342333  9791165342333   
1   9791188700691  9791188700691   
2   9788993267792  9788993267792   
3   9791136248787  9791136248787   
4   9791160023022  9791160023022   
..            ...            ...   
93  9788975278624  9788975278624   
94  9788962021035  9788962021035   
95  9788962020793  9788962020793   
96  9788972207214  9788972207214   
97  9788990220127  9788990220127   

                                            BookTitle BookAuthor Publisher  \
0                      당신의 인생을 정리해드립니다\n삶이 바뀌는 신박한 정리        이지영     쌤앤파커스   
1   아이를 위한 하루 한 줄 인문학 유럽 문화예술 편\n아이의 세계와 시각을 넓혀줄 예...        김종원    청림Life   
2                        자존감 높은 아이로 키우는 애착육아\n[ 개정판 ]    윌리엄 시어스      푸른육아   
3   내 아이에게 들려주는 매일 심리학\n아이의 지성, 사회성, 인성을 키우는 30가지 ...        이동귀       니들북   
4                       엄마의 사랑법\n엄마는 자녀를 어떻게 사랑해야 하는가        장성오     메이트북스   
..                                                ...        ...       ...   
93                           숲과 들을 접시에 담다\n약이 되는 잡초음식        변현단        들녘   
94                         기타부교재\n맛있는 나물\n즐겁고 손쉽게 만드는        유지연       김&정   
95        기타부교재\nLee Vin과 떠나는 즐거운 와인 여행\n3. 프랑스 와인 여행        이동현       김&정   
96                                    야생초차\n산과 들을 마신다        이용성      시골생활   
97                                          부모와 아이 사이   하임 G.기너트       양철북   

          PubDate  Price Category-2  
0   2020년 10월 08일  13500      가정 살림  
1   2020년 10월 07일  14400      가정 살림  
2   2020년 09월 17일  13320      가정 살림  
3   2020년 09월 18일  12150      가정 살림  
4   2020년 09월 17일  13500      가정 살림  
..            ...    ...        ...  
93  2010년 06월 04일  11700      건강 취미  
94  2009년 08월 23일   1940      가정 살림  
95  2009년 04월 20일   1940      가정 살림  
96  2007년 10월 01일  10800      가정 살림  
97  2003년 08월 16일  11700      가정 살림  

[98 rows x 8 columns]

In [4]:
df_interact(book1_1)

interactive(children=(IntSlider(value=0, description='row', max=98, step=7), IntSlider(value=0, description='c…

(98 rows, 8 columns) total


In [5]:
book1_2 = pd.read_csv("Data/book_1_2.csv")
book1_3 = pd.read_csv("Data/book_1_3.csv")
book1_4 = pd.read_csv("Data/book_1_4.csv")
book1_5 = pd.read_csv("Data/book_1_5.csv")
book1_7 = pd.read_csv("Data/book_1_7.csv")
book1_8 = pd.read_csv("Data/book_1_8.csv")
book1_9 = pd.read_csv("Data/book_1_9.csv")
book1_10 = pd.read_csv("Data/book_1_10.csv")
book1_11 = pd.read_csv("Data/book_1_11.csv")
book1_13 = pd.read_csv("Data/book_1_13.csv")
book1_14 = pd.read_csv("Data/book_1_14.csv")
book1_15 = pd.read_csv("Data/book_1_15.csv")
book1_16 = pd.read_csv("Data/book_1_16.csv")
book1_19 = pd.read_csv("Data/book_1_19.csv")
book1_20 = pd.read_csv("Data/book_1_20.csv")
book1_21 = pd.read_csv("Data/book_1_21.csv")
book1_22 = pd.read_csv("Data/book_1_22.csv")
book1_23 = pd.read_csv("Data/book_1_23.csv")
book1_25 = pd.read_csv("Data/book_1_25.csv")
book1_26 = pd.read_csv("Data/book_1_26.csv")
book1_27 = pd.read_csv("Data/book_1_27.csv")

In [7]:
book1_1.append(book1_2).append(book1_3).append(book1_4)

Unnamed: 0           ISBN  \
0   9791165342333  9791165342333   
1   9791188700691  9791188700691   
2   9788993267792  9788993267792   
3   9791136248787  9791136248787   
4   9791160023022  9791160023022   
..            ...            ...   
94  9788957974667  9788957974667   
95  9788956055053  9788956055053   
96  9788962810189  9788962810189   
97  9788971724477  9788971724477   
98  9788934922056  9788934922056   

                                            BookTitle       BookAuthor  \
0                      당신의 인생을 정리해드립니다\n삶이 바뀌는 신박한 정리              이지영   
1   아이를 위한 하루 한 줄 인문학 유럽 문화예술 편\n아이의 세계와 시각을 넓혀줄 예...              김종원   
2                        자존감 높은 아이로 키우는 애착육아\n[ 개정판 ]          윌리엄 시어스   
3   내 아이에게 들려주는 매일 심리학\n아이의 지성, 사회성, 인성을 키우는 30가지 ...              이동귀   
4                       엄마의 사랑법\n엄마는 자녀를 어떻게 사랑해야 하는가              장성오   
..                                                ...              ...   
94                                         영어순해 Basic              김영로   
95                  The Giver 더 기버  [ 교재+워크북+MP3 CD ]           로이스 로리   
96                                      스쿠스쿠 일본어 회화 1  노민영, 카지와라 아야코 공   
97                                        ANIMAL FARM    George Orwell   
98                                    Again! 뒤집어본 영문법              오성호   

             Publisher        PubDate  Price Category-2  
0                쌤앤파커스  2020년 10월 08일  13500      가정 살림  
1               청림Life  2020년 10월 07일  14400      가정 살림  
2                 푸른육아  2020년 09월 17일  13320      가정 살림  
3                  니들북  2020년 09월 18일  12150      가정 살림  
4                메이트북스  2020년 09월 17일  13500      가정 살림  
..                 ...            ...    ...        ...  
94                 넥서스  2011년 01월 25일   8550  국어 외국어 사전  
95               롱테일북스  2010년 12월 20일  13500  국어 외국어 사전  
96  위트앤위즈덤(Wit&Wisdom)  2008년 12월 02일  13500  국어 외국어 사전  
97               반석출판사  2007년 01월 12일   4500  국어 외국어 사전  
98                 김영사  2006년 05월 19일  13410  국어 외국어 사전  

[303 rows x 8 columns]

In [9]:
book1 = book1_1.append(book1_2).append(book1_3).append(book1_4).append(book1_5).append(book1_7).append(book1_8).\
append(book1_9).append(book1_10).append(book1_11).append(book1_13).append(book1_14).append(book1_15).append(book1_16).\
append(book1_19).append(book1_20).append(book1_21).append(book1_22).append(book1_23).append(book1_25).append(book1_26).append(book1_27)

In [10]:
book1

Unnamed: 0           ISBN  \
0   9791165342333  9791165342333   
1   9791188700691  9791188700691   
2   9788993267792  9788993267792   
3   9791136248787  9791136248787   
4   9791160023022  9791160023022   
..            ...            ...   
95  9791163031550  9791163031550   
96  8809254705288  8809254705288   
97  9788965463566  9788965463566   
98  9791188909308  9791188909308   
99  9791188909292  9791188909292   

                                            BookTitle        BookAuthor  \
0                      당신의 인생을 정리해드립니다\n삶이 바뀌는 신박한 정리               이지영   
1   아이를 위한 하루 한 줄 인문학 유럽 문화예술 편\n아이의 세계와 시각을 넓혀줄 예...               김종원   
2                        자존감 높은 아이로 키우는 애착육아\n[ 개정판 ]           윌리엄 시어스   
3   내 아이에게 들려주는 매일 심리학\n아이의 지성, 사회성, 인성을 키우는 30가지 ...               이동귀   
4                       엄마의 사랑법\n엄마는 자녀를 어떻게 사랑해야 하는가               장성오   
..                                                ...               ...   
95       7살 첫 수학 : 시계와 달력 초등 입학 전, 즐거운 공부 기억을 만드는 시간!  강난영, 이은영 /차세정 그림   
96                                     참 쉬운 첫 종이접기 공주               편집부   
97                       강아지 항공사에 찾아온 무시무시한 손님 [ 양장 ]           홍진기 글그림   
98                                     마로네랑 쿠키 [ 양장 ]  토네 사토에 글그림/엄혜숙 역   
99                                     모카랑 핫초코 [ 양장 ]  토네 사토에 글그림/엄혜숙 역   

         Publisher        PubDate  Price Category-2  
0            쌤앤파커스  2020년 10월 08일  13500      가정 살림  
1           청림Life  2020년 10월 07일  14400      가정 살림  
2             푸른육아  2020년 09월 17일  13320      가정 살림  
3              니들북  2020년 09월 18일  12150      가정 살림  
4            메이트북스  2020년 09월 17일  13500      가정 살림  
..             ...            ...    ...        ...  
95  이지스에듀(이지스퍼블리싱)  2020년 04월 10일   7200      초등참고서  
96           애플비북스  2020년 04월 01일   2700         유아  
97            밝은미래  2020년 03월 24일  11700         유아  
98           봄봄출판사  2020년 02월 28일   5400         유아  
99           봄봄출판사  2020년 02월 28일   5400         유아  

[1532 rows x 8 columns]

ISBN이 없는 책(세트 등)

In [12]:
book1[book1["ISBN"] == -1]

Unnamed: 0  ISBN                                          BookTitle  \
46          -1    -1  ETS 토익 정기시험 기출문제집 1000 Vol.2 리스닝 + 리딩 [ 전2권/기출...   
56          -1    -1               영단기 토익 실전 1000제 1 LC + RC 세트 [ 전2권 ]   
65          -1    -1  ETS 토익 정기시험 기출문제집 1000 LISTENING + READING 세트 ...   
66          -1    -1  해커스 토익 실전 1000제 2 Listening + Reading 문제집 세트 [...   
24          -1    -1                          역사의 역사 + 역사란 무엇인가 [ 전2권 ]   
29          -1    -1                              피카파우 동물친구들 세트 [ 전2권 ]   
32          -1    -1                 EBS 고교 1~2학년 수학 기초 패키지 50일 수학 상, 하   
68          -1    -1                           유라시아 견문 1~3권 세트  [ 전3권 ]   
0           -1    -1  전집 사은품증정(랜덤) 큰 별샘 최태성의 초등 별별 한국사 1~4권 세트(전4권)1...   

                                 BookAuthor  Publisher        PubDate  Price  \
46                                      ETS  YBM(와이비엠)  2019년 12월 16일  32040   
56                                  영단기 연구소     커넥츠영단기  2019년 06월 26일   8820   
65                                      ETS  YBM(와이비엠)  2018년 12월 17일  32040   
66                                해커스 어학연구소   해커스어학연구소  2019년 01월 02일  21420   
24                       E.H. 카, 유시민 /김택현 역      YES24  2018년 06월 25일  25200   
29                       얀 쉔켈 /조진경 역/박상숙 감수         참돌  2020년 06월 30일  32400   
32                                      편집부   한국교육방송공사  2016년 12월 15일  13050   
68                                      이병한       서해문집  2019년 01월 10일  58950   
0   최태성  지음 | 김성규,  김차명 (인포그래픽) 그림 | 김택수 감수    엠비씨씨앤아이  2018년 02월 21일  43200   

   Category-2  
46  국어 외국어 사전  
56  국어 외국어 사전  
65  국어 외국어 사전  
66  국어 외국어 사전  
24         역사  
29      건강 취미  
32     중고등참고서  
68         인문  
0          전집

In [13]:
book1.to_csv("Data/book1.csv", sep=',', na_rep='NaN')

분명 카테고리 별로 100권씩 크롤하게 해두었는데 각 카테고리에 속한 책의 수가 일정하지 않았다.
그 카테고리에서 애초에 100권보다 적은 책이 리뷰받았을 가능성과 리뷰 페이지의 카테고리보다 상점 페이지의 카테고리가 세분화되어 있었을 가능성이 있다.

In [18]:
book_grouped_by_category = book1.groupby(['Category-2'])
book_grouped_by_category[["ISBN"]].count()

ISBN
Category-2      
IT 모바일        60
가정 살림         91
건강 취미         78
경제 경영        101
국어 외국어 사전     97
대학교재          29
만화/라이트노벨     101
사회 정치         53
소설/시/희곡        8
수험서 자격증      102
어린이          100
에세이           36
역사            40
예술            67
유아           100
인문            90
자기계발         116
자연과학          34
전집             1
종교            66
중고등참고서       102
청소년           57
초등참고서          2
추천도서           1

## Review 데이터에 대한 분석

review 데이터는 책 데이터와 마찬가지로 22개 카테고리로 나누어서 저장되어 있다. 우선은 그들을 통합한다.

In [20]:
rv1_1 = pd.read_csv("Data/review_1_1.csv")
rv1_2 = pd.read_csv("Data/review_1_2.csv")
rv1_3 = pd.read_csv("Data/review_1_3.csv")
rv1_4 = pd.read_csv("Data/review_1_4.csv")
rv1_5 = pd.read_csv("Data/review_1_5.csv")
rv1_7 = pd.read_csv("Data/review_1_7.csv")
rv1_8 = pd.read_csv("Data/review_1_8.csv")
rv1_9 = pd.read_csv("Data/review_1_9.csv")
rv1_10 = pd.read_csv("Data/review_1_10.csv")
rv1_11 = pd.read_csv("Data/review_1_11.csv")
rv1_13 = pd.read_csv("Data/review_1_13.csv")
rv1_14 = pd.read_csv("Data/review_1_14.csv")
rv1_15 = pd.read_csv("Data/review_1_15.csv")
rv1_16 = pd.read_csv("Data/review_1_16.csv")
rv1_19 = pd.read_csv("Data/review_1_19.csv")
rv1_20 = pd.read_csv("Data/review_1_20.csv")
rv1_21 = pd.read_csv("Data/review_1_21.csv")
rv1_22 = pd.read_csv("Data/review_1_22.csv")
rv1_23 = pd.read_csv("Data/review_1_23.csv")
rv1_25 = pd.read_csv("Data/review_1_25.csv")
rv1_26 = pd.read_csv("Data/review_1_26.csv")
rv1_27 = pd.read_csv("Data/review_1_27.csv")

In [21]:
rv1 = rv1_1.append(rv1_2).append(rv1_3).append(rv1_4).append(rv1_5).append(rv1_7).append(rv1_8).\
append(rv1_9).append(rv1_10).append(rv1_11).append(rv1_13).append(rv1_14).append(rv1_15).append(rv1_16).\
append(rv1_19).append(rv1_20).append(rv1_21).append(rv1_22).append(rv1_23).append(rv1_25).append(rv1_26).append(rv1_27)
rv1

Unnamed: 0                                   Title  ConRate  DesRate  \
0              0            당신의 인생을 정리해드립니다, 내 인생도 정리해보자        5        5   
1              1                         당신의 인생을 정리해드립니다        5        5   
2              2                   당신의 인생을 정리해드립니다 #가정살림        5        5   
3              3                  [가정살림] 당신의 인생을 정리해드립니다        5        4   
4              4                         당신의 인생을 정리해드립니다        5        5   
...          ...                                     ...      ...      ...   
1139        1139  강아지 항공사에 찾아온 무시무시한 손님 / 귀엽고 따뜻한 그림책이에요        5        5   
1140        1140         대 반전 그림책, 강아지 항공사에 찾아온 무시무시한 손님        5        5   
1141        1141                   귀여운 강아지들이 항공사 직원이라니!ㅋ        5        5   
1142        1142                                 마로네랑 쿠키        5        5   
1143        1143                             아이는 좋아하네요 ㅎ        5        5   

       Author        Date                                            Content  \
0     lazyass  2020-10-06  저는 물욕 많고 버릴 줄 모르는 사람이에요. 게다가 정리를 잘 할줄 모르니 어떤 상...   
1     nicebyk  2020-10-06  무조건 남들이 안하는 것특이한 시도를 해보라는 뜻이 아닙니다.구성원의 삶에 초점을 ...   
2        책읽는샘  2020-10-06  2020-145 <당신의 인생을 정리해드립니다(이지영 지음/쌤앤파커스)> #가정살림...   
3        지훈러브  2020-10-06  싱글일때는 나름 깨끗하게 정리를 잘했다고 생각했는데 아이가 태어나고 아이 용품이 늘...   
4     qaz1189  2020-10-06  예전에는 아이들을 모두 어린이집, 유치원에 등원시키고 사부작사부작 거리면서 하나하나...   
...       ...         ...                                                ...   
1139    spqcu  2020-03-31  아이들이 크면 그림책을 안볼 줄 알았는데요특히 저희 둘째는 그림그리기를 좋아해서인지...   
1140       리프  2020-03-31  허웅 ㅠㅠ 너무 귀여워요 ㅠㅠ 어떡해 ㅠㅠ표지부터 완전 심쿵! 세젤귀 강아지들이 오...   
1141  yousoo0  2020-03-25  귀여운 강아지 승무원들이 예쁜 그림체로 등장하는 동화책!깜찍한 항공사 모자를 쓰고있...   
1142  pyr1218  2020-10-06  모카랑 핫초코 책과 함께 구매했는데 책이 너무 작은 사이즈라서 받아보고 놀랐고 제 ...   
1143  pyr1218  2020-10-06  책이 생각보다 진짜 작아 놀랐습니다 ㅎ 아이 보여주기전에 훑어보며 내가 이걸 왜 샀...   

          BookCode  
0     9.791165e+12  
1     9.791165e+12  
2     9.791165e+12  
3     9.791165e+12  
4     9.791165e+12  
...            ...  
1139  9.788965e+12  
1140  9.788965e+12  
1141  9.788965e+12  
1142  9.791189e+12  
1143  9.791189e+12  

[26483 rows x 8 columns]

총 26,483개의 리뷰 데이터가 수집됐다. 이는 참고문헌이었던 *책 리뷰 말뭉치를 활용한 형용사, 부사 감성 사전 개발*의 75만 건보다 굉장히 많이 떨어지는 숫자이다. 참고문헌에서는 각 카테고리에서 리뷰 많은 책 100권으로 크롤했고, 이 분석에서는 지난 일주일간 많이 리뷰된 책 리스트로 크롤했기 때문에 생긴 차이이다.<br>
뒤늦게 알게 되어서 다시 크롤을 진행할 시간적 여유가 있을지 알 수 없다.

데이터프레임에서 index 정보를 지우고, 새로운 인덱스를 지정한 후 csv로 저장한다.

In [30]:
rv1 = rv1.reset_index().drop(["index", "Unnamed: 0"], axis=1)

In [9]:
rv1.to_csv("Data/review1.csv", sep=',', na_rep='NaN')

### 유저별 분석

먼저 추천 필터링에서 가장 큰 의미를 갖는 각 유저 단위에서 분석해보려고 한다.

In [6]:
review_grouped_by_user = rv1.groupby(["Author"])

In [33]:
review_grouped_by_user[["Title"]].count()

Author
Author                        
'_'*                         3
(●’?’●)                      1
**01 22 2020  9:45PM**       3
**03  1 2020  5:03PM**       1
**09  4 2018  1:15PM**       7
...                        ...
히히조아                         1
힐링                           1
힐씨쨩                          1
힘찬발걸음                       10
不狂不及                         1

[11530 rows x 1 columns]

총 11,530 명의 유저가 있는 것을 발견할 수 있었다.<br>
평균적으로 따지면 한 명의 유저당 두 권의 책을 리뷰했다는 것을 알 수 있다. 이 중 한 건의 리뷰만을 써서 유사도를 따질 수 없는 유저들의 리뷰는 삭제한다.

In [7]:
user_group = review_grouped_by_user.groups
user_1_rv = []
for user in user_group.keys() :
    if len(user_group[user]) == 1 :
        user_1_rv.append(user)
len(user_1_rv)

7429

유저 중에서 무려 7429명이 단 한 개의 리뷰만을 작성했다.

In [60]:
rv1[~rv1["Author"].isin(user_1_rv)]

Title  ConRate  DesRate     Author  \
0                당신의 인생을 정리해드립니다, 내 인생도 정리해보자        5        5    lazyass   
2                       당신의 인생을 정리해드립니다 #가정살림        5        5       책읽는샘   
3                      [가정살림] 당신의 인생을 정리해드립니다        5        4       지훈러브   
4                             당신의 인생을 정리해드립니다        5        5    qaz1189   
5                 당신의 인생을 정리해드립니다, 이지영, 쌤앤파커스        5        5        박공주   
...                                       ...      ...      ...        ...   
26476       따뜻한 반전동화! [강아지 항공사에 찾아온 무시무시한 손님]        5        5  lunar1982   
26478  강아지 항공사에 찾아온 무시무시한 손님 / 귀엽고 따뜻한 그림책이에요        5        5      spqcu   
26479         대 반전 그림책, 강아지 항공사에 찾아온 무시무시한 손님        5        5         리프   
26481                                 마로네랑 쿠키        5        5    pyr1218   
26482                             아이는 좋아하네요 ㅎ        5        5    pyr1218   

             Date                                            Content  \
0      2020-10-06  저는 물욕 많고 버릴 줄 모르는 사람이에요. 게다가 정리를 잘 할줄 모르니 어떤 상...   
2      2020-10-06  2020-145 <당신의 인생을 정리해드립니다(이지영 지음/쌤앤파커스)> #가정살림...   
3      2020-10-06  싱글일때는 나름 깨끗하게 정리를 잘했다고 생각했는데 아이가 태어나고 아이 용품이 늘...   
4      2020-10-06  예전에는 아이들을 모두 어린이집, 유치원에 등원시키고 사부작사부작 거리면서 하나하나...   
5      2020-10-06  '당신의 인생을 정리해드립니다.'  저자는 당신의 '집'이 아닌 '인생'을 정리하겠...   
...           ...                                                ...   
26476  2020-04-02  요즘처럼 전국민이 여행에 목말라있는 때가 있었을까 싶을까요?여행의 설렘이 늘 함께하...   
26478  2020-03-31  아이들이 크면 그림책을 안볼 줄 알았는데요특히 저희 둘째는 그림그리기를 좋아해서인지...   
26479  2020-03-31  허웅 ㅠㅠ 너무 귀여워요 ㅠㅠ 어떡해 ㅠㅠ표지부터 완전 심쿵! 세젤귀 강아지들이 오...   
26481  2020-10-06  모카랑 핫초코 책과 함께 구매했는데 책이 너무 작은 사이즈라서 받아보고 놀랐고 제 ...   
26482  2020-10-06  책이 생각보다 진짜 작아 놀랐습니다 ㅎ 아이 보여주기전에 훑어보며 내가 이걸 왜 샀...   

           BookCode  
0      9.791165e+12  
2      9.791165e+12  
3      9.791165e+12  
4      9.791165e+12  
5      9.791165e+12  
...             ...  
26476  9.788965e+12  
26478  9.788965e+12  
26479  9.788965e+12  
26481  9.791189e+12  
26482  9.791189e+12  

[19054 rows x 7 columns]

다음은 많은 리뷰를 단 유저들이 누구인지, 얼마나 많은 리뷰를 달았는지를 살펴보자.

In [35]:
review_grouped_by_user[["Title"]].count().sort_values("Title", ascending=False).head(10)

Title
Author       
샴고양이      295
부자         98
구랭         97
-          95
휘연         91
산바람        78
쿠니토리       73
나날이        67
책찾사        66
박공주        65

다음은 리뷰 점수 분포를 살펴볼 것이다. 간단하게 하기 위해 디자인 점수와 내용점수를 합산할 것이다. 10점 만점이다.

In [63]:
rv1 = rv1[~rv1["Author"].isin(user_1_rv)]
rv1["IntegratedRate"] = rv1["ConRate"] + rv1["DesRate"]
rv1["IntegratedRate"].iplot(kind="hist")

보다시피 10점이 대부분을 차지하고, 8점과 9점이 약간 있으며 그 이하는 거의 없다시피 하다.

---

# 7주차 분석 시작
이번 주에는 못다한 시각화를 몇 가지 하고, Word2Vec 분석에 들어간다.

In [4]:
rv1 = pd.read_csv("Data/review1.csv")

In [12]:
review_grouped_by_book = rv1.groupby("BookCode")
review_grouped_by_book[["Title"]].count().sort_values("Title", ascending=False).iplot(kind="hist")

많은 책의 리뷰수가 10개 미만에 머물렀다. 이것이 분석에 어떤 영향을 미칠지 예의주시해야할 것으로 보인다.

## 형태소 분석